azurerm/resource_arm_key_vault.go

```diff
@@ -52,23 +53,38 @@ func resourceArmKeyVault() *schema.Resource {
 
 			"resource_group_name": resourceGroupNameSchema(),
 
+			// Remove in 2.0
 			"sku": {
-				Type:     schema.TypeList,
-				Required: true,
+				Type:          schema.TypeList,
+				Optional:      true,
+				Deprecated:    "This property has been deprecated in favour of the 'sku_name' property and will be removed in version 2.0 of the provider",
+				ConflictsWith: []string{"sku_name"},
+				MaxItems:      1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"name": {
-							Type:     schema.TypeString,
-							Required: true,
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: suppress.CaseDifference,
 							ValidateFunc: validation.StringInSlice([]string{
 								string(keyvault.Standard),
 								string(keyvault.Premium),
-							}, false),
+							}, true),
 						},
 					},
 				},
 			},
 
+			"sku_name": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
Again, new properties should be case sensitive unless theres good reason

Reply:
Sure...